src/components/Invoice.tsx

```diff
@@ -8,108 +8,171 @@ import Button from '@mui/material/Button';
-            <Button variant="contained" startIcon={<SaveIcon />}>
-              Send
-            </Button>
-          </Grid>
-        </Grid>
-
-        <Grid item container textAlign={'left'} mt={2} spacing={4}>
-          <Grid xs={6} item>
-            <Typography variant="h4" component="p">
-              Recipient
-            </Typography>
-
-            <Box mt={6}>
-              <TextField id="recipient-company-name" label="Company name" variant="standard" fullWidth />
-              <TextField id="recipient-city" label="City" variant="standard" fullWidth />
-              <TextField id="recipient-street" label="Street" variant="standard" fullWidth />
-              <TextField id="recipient-postcode" label="Postcode" variant="standard" fullWidth />
-              <TextField id="recipient-nip" label="NIP" variant="standard" fullWidth />
-              <TextField id="recipient-tel" label="Tel" variant="standard" fullWidth />
-              <TextField id="recipient-email" label="E-mail" variant="standard" fullWidth />
-              <TextField id="recipient-bank-account" label="Bank Account" variant="standard" fullWidth />
-            </Box>
-          </Grid>
-          <Grid xs={6} item>
-            <Typography variant="h4" component="p">
-              Sender
-            </Typography>
-
-            <Box mt={6}>
-              {/* Pytanie Dlaczego czasem daje kursor na koniec */}
-              <TextField id="sender-company-name" label="Company name" variant="standard" fullWidth />
-              <TextField id="sender-city" label="City" variant="standard" fullWidth />
-              <TextField id="sender-street" label="Street" variant="standard" fullWidth />
-              <TextField id="sender-postcode" label="Postcode" variant="standard" fullWidth />
-              <TextField id="sender-nip" label="NIP" variant="standard" fullWidth />
-              <TextField id="sender-tel" label="Tel" variant="standard" fullWidth />
-              <TextField id="sender-email" label="E-mail" variant="standard" fullWidth />
-              <TextField id="sender-bank-account" label="Bank Account" variant="standard" fullWidth />
-            </Box>
-          </Grid>
-        </Grid>
 
-        <Grid item container>
-          <Grid item container spacing={3}>
-            <Grid item xs={6}>
-              <TextField id="name" label="Name" variant="standard" fullWidth />
-            </Grid>
+          <Grid item container textAlign={'left'} mt={2} spacing={4}>
+            <Grid xs={6} item>
+              <Typography variant="h4" component="p">
+                Recipient
+              </Typography>
 
-            <Grid item xs={1.2}>
-              <TextField type="number" id="amount" label="Amount" variant="standard" fullWidth />
+              <Box mt={6}>
+                <TextField {...register('recipient.companyName', { required: true })} id="recipient-company-name" label="Company name" variant="standard" fullWidth />
```

Comment:
The code here looks pretty repetitive. Do you think you can create `PersonalDataForm` component or something, and reuse it both, for `recipient` and `sender`?

Reply:
Created, but somethings is not working. Form stopped to be sended

Reply:
done